grpc/params.go

```diff
@@ -99,3 +118,130 @@ func (p *callParams) SetSystemTags(state *lib.State, addr string, methodName str
 		p.TagsAndMeta.SetSystemTagOrMetaIfEnabled(state.Options.SystemTags, metrics.TagName, methodName)
 	}
 }
+
+// connectParams is the parameters that can be passed to a gRPC connect call.
+type connectParams struct {
+	IsPlaintext           bool
+	UseReflectionProtocol bool
+	ReflectionMetadata    metadata.MD
+	Timeout               time.Duration
+	MaxReceiveSize        int64
+	MaxSendSize           int64
+	TLS                   map[string]interface{}
+}
+
+func newConnectParams(vu modules.VU, input goja.Value) (*connectParams, error) { //nolint:gocognit
+	result := &connectParams{
+		IsPlaintext:           false,
+		UseReflectionProtocol: false,
+		Timeout:               time.Minute,
+		MaxReceiveSize:        0,
+		MaxSendSize:           0,
+		ReflectionMetadata:    metadata.New(nil),
+	}
+
+	if common.IsNullish(input) {
+		return result, nil
+	}
+
+	rt := vu.Runtime()
+	params := input.ToObject(rt)
+
+	for _, k := range params.Keys() {
+		v := params.Get(k).Export()
+
+		switch k {
+		case "plaintext":
+			var ok bool
+			result.IsPlaintext, ok = v.(bool)
+			if !ok {
+				return result, fmt.Errorf("invalid plaintext value: '%#v', it needs to be boolean", v)
+			}
+		case "timeout":
+			var err error
+			result.Timeout, err = types.GetDurationValue(v)
+			if err != nil {
+				return result, fmt.Errorf("invalid timeout value: %w", err)
+			}
+		case "reflect":
+			var ok bool
+			result.UseReflectionProtocol, ok = v.(bool)
+			if !ok {
+				return result, fmt.Errorf("invalid reflect value: '%#v', it needs to be boolean", v)
+			}
+		case "reflectMetadata":
+			md, err := newMetadata(params.Get(k))
+			if err != nil {
+				return result, fmt.Errorf("invalid metadata param: %w", err)
+			}
+
+			result.ReflectionMetadata = md
```

Comment:
nit: This as well as the error messages in the `newMetadata` refer to `metadata` while the property here is `reflectMetadata` and it might confuse users.

Reply:
Right, let me fix that